aiida/orm/querybuilder.py

```diff
@@ -2006,46 +2051,33 @@ def get_used_tags(self, vertices: bool = True, edges: bool = True) -> List[str]:
                 given_tags.append(path['edge_tag'])
         return given_tags
 
-    def get_query(self):
-        """
-        Instantiates and manipulates a sqlalchemy.orm.Query instance if this is needed.
-        First,  I check if the query instance is still valid by hashing the queryhelp.
-        In this way, if a user asks for the same query twice, I am not recreating an instance.
+    def get_query(self) -> 'Query':
+        """Return the sqlalchemy.orm.Query instance for the current query specification.
 
-        :returns: an instance of sqlalchemy.orm.Query that is specific to the backend used.
+        To avoid unnecessary re-builds of the query, the hashed dictionary representation of this instance
+        is compared to the last query returned, which is cached by its hash.
         """
         from aiida.common.hashing import make_hash
 
-        # Need_to_build is True by default.
-        # It describes whether the current query
-        # which is an attribute _query of this instance is still valid
-        # The queryhelp_hash is used to determine
-        # whether the query is still valid
-
-        queryhelp_hash = make_hash(self.queryhelp)
-        # if self._hash (which is None if this function has not been invoked
-        # and is a string (hash) if it has) is the same as the queryhelp
-        # I can use the query again:
-        # If the query was injected I never build:
```

Comment:
Bad joke incoming ...


As we forget faster and faster and -- what?
Oh right! Maybe put this in the doc string of the method instead of removing completely, to keep some of the humanly laid out logic behind the code.

Reply:
I felt these comments were just describing what you can already easily see from the code and not adding any actual extra insight.

Reply:
I agree to an extent. I think the purpose of having the hash is not self-explanatory from the code though, unless you are _very_ into the code and always use hashes this way - and even then, I think it's fair to put in a human-readable comment either by the `make_hash` line or in the doc-string to quickly mention the purpose (i.e., reusing existing `QueryBuilder` instances versus making new ones, I guess?) and that by default the value in the following lines is `None`, something else that might not be clear from the code, since it only relays what to do _if_ the value is `None`.

Reply:
It is already explained in the docstring, reagrding the use of a hash to avoid rebuilds, I've added an extra comment below though about if the hash is None